src/glmfit.jl

```diff
@@ -273,6 +273,10 @@ end
 
 dof(x::GeneralizedLinearModel) = dispersion_parameter(x.rr.d) ? length(coef(x)) + 1 : length(coef(x))
 
+dof(obj::GeneralizedLinearModel{<:GlmResp,<:DensePredChol{<:Real,<:CholeskyPivoted}}) = dispersion_parameter(obj.rr.d) ? obj.pp.chol.rank + 1 : obj.pp.chol.rank
```

Comment:
Defining this method for `CholeskyPivoted` here breaks the organization of internal interfaces. Better define `rank(::LinPred)` in linpred.jl for each `LinPred` type and call it here instead of `length(coef(x))` to get the rank for any `GeneralizedLinearModel`.

Reply:
Okay.

Reply:
I tried defining a `rank` function, which is conflict with the rank function of linear algebra. So added a `model_rank`.
I am sure there might be a way to overload the `rank` function. Let me know your view.

Reply:
`model_rank` is OK. But I was thinking about defining `rank(::LinPred)`, not `rank(::GeneralizedLinearModel)`. Otherwise it doesn't allow organizing interfaces more logically (by defining `rank(::LinPred)` in linpred.jl).

Reply:
To be clearer: I think you can drop `model_rank(::GeneralizedLinearModel)` and `model_rank(::LinearModel)` and just call `model_rank(x.pp)` directly instead.

BTW, thinking about it, maybe `linpred_rank` would be more appropriate than `model_rank`? Or just override the `LinearAlgebra` function using `LinearAlgebrarank(::LinearModel) = ...`.